src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
children.forEach((ch) => {
    const mother = mothers[ch.mother];
    if (mother) {
      sum += ch.born - mother.born;
    }
  });

Reply:
Checking for the presence of the mother: The code does not take into account the case when the mother is not found in the people array. This can cause an error when we try to get the .born property of a non-existent object. It is recommended to add a check for the presence of the mother before accessing the properties.